File: src/main/java/org/blacksun/graph/channel/ChannelFactory.java
package org.blacksun.graph.channel;

import org.blacksun.graph.node.GraphNode;
import org.jetbrains.annotations.NotNull;

public interface ChannelFactory {
    default Channel createChannel(@NotNull GraphNode from, @NotNull GraphNode to, int weight) {
        return createChannel(from, to, weight, Channel.ERRORS_AMOUNT);
    }

    Channel createChannel(@NotNull GraphNode from, @NotNull GraphNode to, int weight, double errors);
}

File: src/main/java/org/blacksun/graph/channel/ChannelFactory.kt
package org.blacksun.graph.channel

import org.blacksun.graph.node.GraphNode

interface ChannelFactory {
    fun createChannel(from: GraphNode, to: GraphNode, weight: Int): Channel {
        return createChannel(from, to, weight, Channel.ERRORS_AMOUNT)
    }

    fun createChannel(from: GraphNode, to: GraphNode, weight: Int, errors: Double): Channel
}
